applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp

```diff
@@ -21,203 +19,802 @@
-            .def("SetKnotU", &Type::SetKnotU,
-                "Index"_a,
-                "Value"_a)
-            .def("SetKnotV", &Type::SetKnotV,
-                "Index"_a,
-                "Value"_a)
-            .def("Node", &Type::Node,
-                "IndexU"_a,
-                "IndexV"_a)
-            .def("SetNode", &Type::SetNode,
-                "IndexU"_a,
-                "IndexV"_a,
-                "Value"_a)
-            .def("Pole", &Type::Pole,
-                "IndexU"_a,
-                "IndexV"_a)
-            .def("SetPole", &Type::SetPole,
-                "IndexU"_a,
-                "IndexV"_a,
-                "Value"_a)
-            .def("Weight", &Type::Weight,
-                "IndexU"_a,
-                "IndexV"_a)
-            .def("SetWeight", &Type::SetWeight,
-                "IndexU"_a,
-                "IndexV"_a,
-                "Value"_a)
-            .def("PointAt", &Type::PointAt,
-                "U"_a,
-                "V"_a)
-            .def("DerivativesAt", &Type::DerivativesAt,
-                "U"_a,
-                "V"_a,
-                "Order"_a)
-            .def("ValueAt", (double (Type::*)(const Variable<double>&,
-                const double, const double) const) &Type::ValueAt<double>,
-                "Variable"_a,
-                "U"_a,
-                "V"_a)
-            .def("ValueAt", (std::vector<double> (Type::*)(
-                const Variable<double>&, const double, const double, const int)
-                const)
-                &Type::ValueAt<double>,
-                "Variable"_a,
-                "U"_a,
-                "V"_a,
-                "Order"_a)
-            .def("ValueAt", (Array3D (Type::*)(const Variable<Array3D>&,
-                const double, const double) const) &Type::ValueAt<Array3D>,
-                "Variable"_a,
-                "U"_a,
-                "V"_a)
-            .def("ValueAt", (std::vector<Array3D> (Type::*)(
-                const Variable<Array3D>&, const double, const double,
-                const int) const) &Type::ValueAt<Array3D>,
-                "Variable"_a,
-                "U"_a,
-                "V"_a,
-                "Order"_a)
-            .def("ValueAt", (double (Type::*)(const VariableComponent&,
-                const double, const double) const) &Type::ValueAt<double,
-                VariableComponent>,
-                "Variable"_a,
-                "U"_a,
-                "V"_a)
-            .def("ValueAt", (std::vector<double> (Type::*)(
-                const VariableComponent&, const double, const double, const int)
-                const) &Type::ValueAt<double, VariableComponent>,
-                "Variable"_a,
-                "U"_a,
-                "V"_a,
-                "Order"_a)
-        ;
-    }
+void RegisterPoint1D(
+    pybind11::module& m,
+    const std::string& name)
+{
+    namespace py = pybind11;
+    using namespace pybind11::literals;
+
+    using Type = Kratos::array_1d<double, 1>;
```

Comment:
calling it "type" is quite generic. Can't u call IntervalType or something of the sort?

i am speaking thinking of future naming conflicts (i do appreciate that Type is local to the function, so ignore this comment if you really dislike it)

Reply:
The idea is that there is only one `Type` alias per `Register<Type>` function.

But there is the risk of naming conflicts. Types like `Interval` or `Curve2D` might be also declared in other applications. Not at the moment but maybe in the future. So we should think about using namespaces, using prefixes, moving stuff into the core,...